polars/polars-io/src/csv/read.rs

```diff
@@ -383,6 +383,12 @@ impl<'a, R: MmapBytesReader + 'a> CsvReader<'a, R> {
                     fld.coerce(DataType::Int32);
                     Some(fld)
                 }
+                Int32 | Int64 | UInt32 | UInt64 => {
```

Comment:
Why do these need to be cast? We have have native support for these.

Reply:
I may have altered something via the match arms, but what seems to be happening is that if either of `self.schema_overwrite` or `self.dtype_overwrite` are set, then `self.prepare_schema_overwrite(schema)` is used which, in turn, uses the casting. Without this (superfluous) cast in here, I couldn't seem to get the buffer to read the dtype as specified. I'll look again though.